tests/test_objective_funs.py

```diff
@@ -816,6 +819,39 @@ def test(coil, grid=None):
         test(mixed_coils, grid=[LinearGrid(N=5)] * len(mixed_coils.coils))
         test(nested_coils, grid=nested_grids)
 
+    @pytest.mark.unit
+    def test_toroidal_flux(self):
+        """Test calculation of toroidal flux from coils."""
+        grid1 = LinearGrid(L=10, M=10, zeta=np.array(0.0))
+
+        def test(eq, field, correct_value, rtol=1e-14, grid=None):
+            obj = ToroidalFlux(eq=eq, field=field, eval_grid=grid)
+            obj.build(verbose=2)
+            torflux = obj.compute_unscaled(*obj.xs(field))
+            np.testing.assert_allclose(torflux, correct_value, rtol=rtol)
+
+        eq = Equilibrium(iota=PowerSeriesProfile(0))
+        test(eq, VerticalMagneticField(B0=1), 0, grid=grid1)
```

Comment:
I'm confused by this test. The equilibrium is a purely toroidal field, but you are comparing to a purely vertical field? I would expect the equilibrium to have a flux of Psi=1 (the default) and the field to have Psi=0. So why is the "correct value" here 0?

Reply:
I am testing the objective function, which calculates the toroidal flux from the given magnetic field. The given magnetic field here is a pure vertical field, thus it generates zero toroidal flux. This just is checking that the objective correctly calculates the toroidal flux as zero.